tests/utilities.py

```diff
@@ -103,4 +106,64 @@ def rerun_marqo_with_env_vars(env_vars: list = [], calling_class: str = ""):
 def rerun_marqo_with_default_config(calling_class: str = ""):
     # Do not send any env vars
     # This should act like running the start script at the beginning
-    rerun_marqo_with_env_vars(env_vars=[], calling_class=calling_class)
\ No newline at end of file
+    rerun_marqo_with_env_vars(env_vars=[], calling_class=calling_class)
+
+
+docker_log_failure_message = "Failed to fetch docker logs for Marqo"
+
+
+def attach_docker_logs(container_name: str, log_collection: typing.List) -> None:
+    """Fetches the Docker logs of a specified container and stores them in a provided list.
+
+    Args:
+        container_name (str): Name of the Docker container whose logs are to be
+            fetched.
+        log_collection (List): A list which its first element is used to store
+            the fetched logs or error messages. A list is used to store the
+            logs, rather than returning them, so this function can be used in a
+            thread.
+    """
+    completed_process = subprocess.run(
+        ["docker", "logs", container_name],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        text=True
+    )
+
+    if completed_process.returncode == 0:
+        log_collection.append(completed_process.stdout)
+    else:
+        log_collection.append(
+            f"{docker_log_failure_message}. "
+            f"Failed with error: {completed_process.stderr}")
+
+
+def retrieve_docker_logs(
+        container_name: str
+) -> str:
+    """Returns docker logs as a string, for a specific container
+
+    Args:
+        container_name (str): Name of the Docker container whose logs are to be checked. Defaults to 'marqo'.
+    Returns:
+        A str which is the docker logs for the container.
+    Raises:
+        RuntimeError: If fetching logs fails or times out.
+    """
+    # a 1-elem mutable object to save the docker logs to:
+    log_collection = []
+
+    docker_log_fetcher = attach_docker_logs
+
+    # Run the fetch_docker_logs function in a separate thread
```

Comment:
Thanks Joshua. I removed the outdated comment. 

I prefer keeping `docker_log_fetcher` as a noun, because it is  used as a variable in this function